addons/knobs/src/components/Panel.js

```diff
@@ -135,7 +136,18 @@ export default class KnobPanel extends PureComponent {
       ...changedKnob,
     };
 
-    this.setState({ knobs: newKnobs }, this.emitChange(changedKnob));
+    this.setState({ knobs: newKnobs }, () => {
+      this.emitChange(changedKnob);
+
+      const queryParams = {};
+
+      Object.keys(newKnobs).forEach(n => {
+        const knob = newKnobs[n];
+        queryParams[`knob-${n}`] = Types[knob.type].serialize(knob.value);
+      });
+
+      api.setQueryParams(queryParams);
+    });
```

Comment:
I did see these code before, are they duplicated?!

Reply:
Yes, it is in `Panel.setknobs` & `Panel.handleChange`, I don't want to refactor knobs right now, considering this is for the 5.1 release,

But this duplication would be nice to remove.